File: android/src/main/java/com/abelium/cardvalidator/CardNumberMatch.java
package com.abelium.cardvalidator;

public class CardNumberMatch
{
    public static final CardNumberMatch NO_MATCH = new CardNumberMatch(null, Validity.Invalid);
    public static final CardNumberMatch EMPTY = new CardNumberMatch(null, Validity.Partial);

    private CreditCardType cardType;
    private Validity validity;

    public CardNumberMatch(CreditCardType cardType, Validity match) {
        this.cardType = cardType;
        this.validity = match;
    }

    public CreditCardType getCardType() {
        return cardType;
    }

    public Validity getValidity() {
        return validity;
    }
}

File: ios/RCTBraintree/CCValidator/CardNumberMatch.m
//
//  CardNumberMatch.m
//  RCTBraintree
//
//  Created by Urska Pangerc on 28/11/2016.
//  Copyright © 2016 Rickard Ekman. All rights reserved.
//

#import "CardNumberMatch.h"
#import "Validity.h"

@implementation CardNumberMatch

//CardNumberMatch * const NO_MATCH = [[CardNumberMatch alloc] initWithCreditCardType:nil withValidity:Invalid];
//CardNumberMatch * const EMPTY =[[CardNumberMatch alloc] initWithCreditCardType:nil withValidity:Partial];

-(id)initWithCreditCardType:(CreditCardType *)cardType withValidity:(Validity *) match
{
    self = [super init];
    if(self)
    {
        self.cardType = cardType;
        self.validity = match;
    }
    return self;
}

-(CreditCardType *)getCardType
{
    return _cardType;
}

-(Validity)getValidity
{
    return _validity;
}


@end

